lib/cadet/course/course.ex

```diff
@@ -57,6 +56,28 @@ defmodule Cadet.Course do
     end
   end
 
+  @doc """
+  Create group entity with given name
+  """
+  def create_group(name, leader, mentor) do
+    %Group{}
+    |> Group.changeset(%{name: name})
+    |> change(%{leader_id: Map.get(leader, :id)})
+    |> change(%{mentor_id: Map.get(mentor, :id)})
+    |> Repo.insert()
+  end
+
+  @doc """
+  Adds given student to given group
+  """
+  def add_student_to_group(group = %Group{}, student = %User{role: :student}) do
+    student
+    |> Repo.preload(:group)
+    |> User.changeset()
+    |> change(%{group_id: Map.get(group, :id)})
```

Comment:
You should create a new `User.student_changeset/2` that takes in `:group_id` as well. Again, look at how the `Assessments` context does this.